packages/web/src/utils/index.js

```diff
@@ -130,3 +130,17 @@ export function parseValueArray(originalArr = [], currentValue) {
 export function escapeRegExp(string) {
 	return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
 }
+
+/**
+ * @param {Object} props
+ * @param {Object} prevProps
+ * @param {'defaultQuery' | 'customQuery'} key
+ */
+export const isQueryIdentical = (props = {}, prevProps = {}, key) => {
+	if (!key) return true;
+	if (typeof props[key] !== 'function' || typeof prevProps[key] !== 'function') return true;
+	// to not call original defaultQuery and customQuery, as here we are only comparing
+	const prevQuery = () => prevProps[key];
+	const nextQuery = () => props[key];
+	return isEqual(nextQuery(props.value, props), prevQuery(prevProps.value, prevProps));
```

Comment:
@ShahAnuj2610 You'll not find `value` in the props, you need to pass the current value from the component.

Reply:
@bietkul what would happen in case of a controlled component?

Reply:
@ShahAnuj2610 Just use the `value` from the state.
Check the logic where we evaluate the `customQuery` in `updateQuery` method.

Reply:
okay, doing it.

Reply:
also @bietkul since we're comparing queries, it would be compared wrt to  **this.state.currentValue** and not **prevState.currentValue**.

E.g. it would be 

```
isEqual(
nextQuery(this.state.currentValue, props), 
prevQuery(this.state.currentValue, prevProps)
);
```

and not

```
isEqual(
nextQuery(this.state.currentValue, props), 
prevQuery(prevState.currentValue, prevProps)
);
```

Reply:
@bietkul have fixed it